backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -691,6 +682,137 @@ public void validateDiffMergeRequest(
+
+      if (!VersioningUtils.commitRepositoryMappingExists(
+          session, request.getCommitToRevertSha(), repositoryEntity.getId())) {
+        throw new ModelDBException(
+            "No such revert commit found in the repository : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
+      }
+
+      if (!VersioningUtils.commitRepositoryMappingExists(
+          session, request.getBaseCommitSha(), repositoryEntity.getId())) {
+        throw new ModelDBException(
+            "No such base commit found in the repository : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
+      }
+
+      commitToRevertShaEntity = session.get(CommitEntity.class, request.getCommitToRevertSha());
+      baseCommitEntity = session.get(CommitEntity.class, request.getBaseCommitSha());
+
+      if (commitToRevertShaEntity.getParent_commits() == null
+          || commitToRevertShaEntity.getParent_commits().isEmpty()) {
+        throw new ModelDBException(
+            "No parent found for commit : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
+      }
+      CommitEntity firstParentOfCommitToRevert =
+          new ArrayList<>(commitToRevertShaEntity.getParent_commits()).get(0);
+
+      Map<String, Map.Entry<BlobExpanded, String>> locationBlobsMapFirstParentCommit =
+          getCommitBlobMapWithHash(
+              session, firstParentOfCommitToRevert.getRootSha(), new ArrayList<>());
+
+      Map<String, Map.Entry<BlobExpanded, String>> locationBlobsMapBaseCommit =
+          getCommitBlobMapWithHash(session, baseCommitEntity.getRootSha(), new ArrayList<>());
+
+      Map<String, Map.Entry<BlobExpanded, String>> locationBlobsMapCommitToRevert =
+          getCommitBlobMapWithHash(
+              session, commitToRevertShaEntity.getRootSha(), new ArrayList<>());
+
+      List<ai.verta.modeldb.versioning.BlobDiff> commitToRevertShaBlobDiff =
+          computeDiffFromCommitMaps(
+                  locationBlobsMapCommitToRevert, locationBlobsMapFirstParentCommit)
+              .getDiffsList();
+      Map<String, BlobExpanded> locationBlobsMapBaseCommitSimple =
+          convertToLocationBlobMap(locationBlobsMapBaseCommit);
+
+      blobContainerList =
+          getBlobContainers(
+              commitToRevertShaBlobDiff.stream()
+                  .map(AutogenBlobDiff::fromProto)
+                  .collect(Collectors.toList()),
+              locationBlobsMapBaseCommitSimple,
+              new LinkedHashMap<>());
+    }
+
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Transaction transaction = session.beginTransaction();
+      List<String> parentSHAs = Collections.singletonList(request.getBaseCommitSha());
+      List<CommitEntity> parentCommits = Collections.singletonList(baseCommitEntity);
+      String revertMessage = request.getContent().getMessage();
+      if (revertMessage.isEmpty()) {
+        revertMessage =
```

Comment:
do a syntaxtically correct version of 
`revertMessage = VersioningUtils.revertCommitMessage(commitToRevertShaEntity.getCommitMessage());`

Reply:
Fixed